### PR TITLE
Adding wolf breading and moving breeding functionality to cMonster

### LINKED ISSUE #4911: Wolf breeding is not possible
Cuberite Commit id: 23bca008317e125659d4b97f29cf13da81a7aecd

### Expected behavior
tamed Wolfes can be fed with any kind of meat to bread and spawn a new wolf

### Actual behavior
No food is accepted when fully healed

### Steps to reproduce the behavior
- Spawn wolf
- tame it
- try to feed it

### PR DESCRIPTION
**Pull request by 12xx12**
I moved the breeding code to cMonster.

I tried adding a class that can be a ancestor for breedables but that failed.
So now any monster can be bread :D

Fixes #4911

### CODE REVIEW

**Comment by 12xx12** [comment]
Whoops this PR went from 0 to 100 realy fast

**Comment by 12xx12** [comment]
I'm not quite shure if we want to keep it like that. It's just an experiment

**Comment by 12xx12** [comment]
Breedable was a bad idea... moving the breeding functionality to cMonster

File: src/Mobs/Wolf.cpp [hunk 1]
```diff
@@ -337,6 +372,8 @@ void cWolf::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	{
 		StopMovingToPosition();
 	}
+
+	cMonster::LoveTick();
```
Comment (bearbin) [inline src/Mobs/Wolf.cpp:376]: This needs a bit of testing - this way loving takes priority over all other behaviours, is that correct in vanilla? It could be that LoveTick needs to be moved more to the start of the function.

File: src/Mobs/Wolf.cpp [hunk 2]
```diff
@@ -16,7 +17,6 @@ cWolf::cWolf(void) :
 	m_IsTame(false),
 	m_IsBegging(false),
 	m_IsAngry(false),
-	m_OwnerName(""),
```
Comment (bearbin) [inline src/Mobs/Wolf.cpp:19]: Why was this removed?

Comment (12xx12) [reply src/Mobs/Wolf.cpp:19]: my local clang said that this is redundant string initialization

File: src/Mobs/Wolf.cpp (hunk 1 above)
Comment (12xx12) [reply src/Mobs/Wolf.cpp:376]: I did some testing...

The behaviour is a bit strage:

The wolfes run into each other and then attack the enemy and then run into each other a create an offspring.

**Comment by 12xx12** [comment]
This implementation also derivates from notchian implementation in that way that the wolves chase you if you have their food in your hand. not the respectful distance as ausual but bump into you like every other feedable

File: src/Mobs/Wolf.cpp [hunk 3]
```diff
@@ -231,6 +246,11 @@ void cWolf::OnRightClicked(cPlayer & a_Player)
 				}
 				break;
 			}
+			// multiplication is handheld in cMonster. Just prevents from sitting down.
```
Comment (bearbin) [inline src/Mobs/Wolf.cpp:249]: Now a typo replaced with a typo :P

It should be `handled`

File: src/Mobs/Monster.cpp [hunk 4]
```diff
@@ -14,13 +14,15 @@
...
+#include "Items/ItemSpawnEgg.h"
+
 #include "../Chunk.h"
 #include "../FastRandom.h"
 
 #include "PathFinder.h"
 #include "../Entities/LeashKnot.h"
 
-
+const int TPS = 20;
```
Comment (bearbin) [inline src/Mobs/Monster.cpp:25]: This should actually go in one of the global headers, I'm rather surprised we don't have such a constant already, but Monster.cpp isn't really the right place for it.

Comment (12xx12) [reply src/Mobs/Monster.cpp:25]: ```Defines.h``` maybe ?

File: src/Mobs/Monster.cpp [hunk 5]
```diff
@@ -1248,6 +1258,195 @@ std::unique_ptr<cMonster> cMonster::NewMonsterFromType(eMonsterType a_MobType)
...
+		}
+	}
+	else
+	{
+		// We have no partner, so we just chase the player if they have our breeding item
+		cItems FollowedItems;
+		GetFollowedItems(FollowedItems);
+		if (FollowedItems.Size() > 0)
+		{
+			m_World->DoWithNearestPlayer(GetPosition(), static_cast<float>(m_SightDistance), [&](cPlayer & a_Player) -> bool
```
Comment (NiLSPACE) [inline src/Mobs/Monster.cpp:1339]: Eventually we probably should create a `DoWithNearestPlayers` which would loop the players in order of the distance between the specified point. Now if the nearest player doesn't have the `FollowedItem` equiped, but the next player does he is ignored.

**Review by bearbin — APPROVED** [review]
The build fails, but once you get it to pass, this LGTM.

**Comment by 12xx12** [comment]
Some error with a lua state file...

I updated the branch to master let's see what happens

**Comment by 12xx12** [comment]
What did I break this time?

**Comment by bearbin** [comment]
This time it's tolua failing. No idea why though

**Comment by Seadragon91** [comment]
Looks like the change from commit `moved tps to Defines.h` causes that problem.

**Comment by bearbin** [comment]
Maybe try not exporting TPS to the lua API. It makes sense to export it, but not if it causes the whole system to break for no reason.

**Comment by tigerw** [comment]
Best to tag TPS as `inline constexpr` so the linker gives it one address only

**Comment by 12xx12** [comment]
@tigerw I tried adding your suggestion, But It seems to break the build

**Comment by tigerw** [comment]
goodness me. Wrap it in a
```
#ifdef __clang__
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wmissing-variable-declarations"
#endif
// ...
#ifdef __clang__
#pragma clang diagnostic pop
#endif
```
I can fix arm64 later.

**Comment by 12xx12** [comment]
This seems kinda scetchy - but ok

**Comment by tigerw** [comment]
arm64 is on clang6, it needs to be upgraded to clang7 to silence the warning

**Comment by 12xx12** [comment]
that makes sense 💡
